libcontainer/cgroups/fs/devices.go

```diff
@@ -26,7 +26,11 @@ func (s *DevicesGroup) Apply(d *cgroupData) error {
 }
 
 func (s *DevicesGroup) Set(path string, cgroup *configs.Cgroup) error {
-	if system.RunningInUserNS() {
+	pids, err := cgroups.GetPids(path)
+	if err != nil {
+		return err
+	}
+	if system.RunningInUserNS(pids[0]) {
```

Comment:
A better way to do it is to use the pids field in `configs.Cgroup` -- at this point we only have one PID joining the cgroup so it's a little weird to use `GetPids(path)[0]`.